pkg/store/database.go

```diff
@@ -6,36 +6,63 @@ import (
 	"github.com/nrc-no/core/pkg/api/meta"
 	"github.com/nrc-no/core/pkg/api/types"
 	"github.com/nrc-no/core/pkg/sql/convert"
-	uuid "github.com/satori/go.uuid"
 	"go.uber.org/zap"
 	"gorm.io/gorm"
 	"time"
 )
 
+// DatabaseStore is the store for types.Database
 type DatabaseStore interface {
+	// Get a Database by ID
 	Get(ctx context.Context, databaseID string) (*types.Database, error)
+	// List databases
 	List(ctx context.Context) (*types.DatabaseList, error)
+	// Create a database
 	Create(ctx context.Context, database *types.Database) (*types.Database, error)
+	// Delete a database
 	Delete(ctx context.Context, databaseID string) error
 }
 
+// NewDatabaseStore returns a new DatabaseStore
 func NewDatabaseStore(db Factory) DatabaseStore {
-	return &databaseStore{db: db}
+	return &databaseStore{
+		db:             db,
+		deleteDatabase: convert.DeleteDatabaseSchemaIfExist,
+		createDatabase: convert.CreateDatabase,
+	}
 }
 
+// Database is the store model for types.Database
 type Database struct {
-	ID        string
-	Name      string
+	// ID corresponds to the types.Database ID
+	ID string
+	// Name corresponds to the types.Database Name
+	Name string
+	// CreatedAt corresponds to the types.Database CreatedAt
 	CreatedAt time.Time
+	// UpdatedAt corresponds to the types.Database UpdatedAt
 	UpdatedAt time.Time
 }
 
+type deleteDatabaseFn func(db *gorm.DB, databaseId string) error
+type createDatabaseFn func(db *gorm.DB, database *types.Database) error
+
+// databaseStore is the implementation of DatabaseStore
 type databaseStore struct {
+	// db is the database Factory
 	db Factory
+	// deleteDatabase is a function for deleting the backing sql database
```

Comment:
what does backing mean?

Reply:
when we create an object `Database`, it creates an actual `SQL Database` on the database server...

Reply:
could we say "actual" instead of "backing" then? "backing" sounds like "supporting" to me

Reply:
done :)